airflow/executors/base_executor.py

```diff
@@ -185,25 +221,22 @@ def get_event_buffer(self, dag_ids=None):
         return cleared_events
 
     def execute_async(self,
-                      key,
-                      command,
-                      queue=None,
-                      executor_config=None):  # pragma: no cover
+                      key: TaskInstanceKey,
+                      command: CommandType,
+                      executor_config: Optional[Any] = None) -> None:  # pragma: no cover
         """
         This method will execute the command asynchronously.
+
+        :param key: Unique key for the task instance
+        :param command: Command to run
+        :param executor_config: Configuration passed to the executor.
         """
         raise NotImplementedError()
 
-    def end(self):  # pragma: no cover
+    def end(self) -> None:  # pragma: no cover
         """
         This method is called when the caller is done submitting job and
         wants to wait synchronously for the job submitted previously to be
         all done.
         """
         raise NotImplementedError()
-
-    def terminate(self):
```

Comment:
Also feels like an API change.

Reply:
Again - not used. But if you think Executors are part of the "public" API, then I can add it back - no problem.

Reply:
The problem was that none of the executors implemented it so I removed it to get rid of "Do not implement all abstract methods" warning.... I will add an empty terminate() method to all the current executors.